src/main/java/com/aws/greengrass/clientdevices/auth/api/UseCases.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.api;
+
+import com.aws.greengrass.dependency.Context;
+import com.aws.greengrass.util.CrashableFunction;
+
+
+public class UseCases {
+    private final Context context;
+    private static UseCases instance;
```

Comment:
I will caution you against statics/singleton pattern. It makes testing and dependency injection more of a hassle (if not impossible).

Reply:
Ironically this actually makes it much easier to use DI throughout the rest of the code